File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/reports/components/MMReportReadinessChecklist.kt
package com.surensureshkumar.movemark.features.reports.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.outlined.Circle
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.surensureshkumar.movemark.core.design.MMColors
import com.surensureshkumar.movemark.core.design.mmAppearRise
import com.surensureshkumar.movemark.core.design.components.MMProofCard

enum class ChecklistItemState { Complete, Incomplete, Locked }

data class ReportChecklistItem(
    val title: String,
    val detail: String,
    val state: ChecklistItemState,
)

@Composable
fun MMReportReadinessChecklist(
    items: List<ReportChecklistItem>,
    appeared: Boolean,
    reduceMotion: Boolean,
    modifier: Modifier = Modifier,
) {
    Column(modifier) {
        Text(
            "Proof checklist",
            color = MMColors.TextMuted,
            fontSize = 13.sp,
            fontWeight = FontWeight.SemiBold,
            modifier = Modifier.padding(start = 2.dp, bottom = 10.dp),
        )
        MMProofCard {
            items.forEachIndexed { index, item ->
                if (index > 0) {
                    HorizontalDivider(
                        modifier = Modifier.padding(vertical = 6.dp),
                        color = Color.White.copy(alpha = 0.06f),
                    )
                }
                ChecklistRow(
                    item = item,
                    modifier = Modifier.mmAppearRise(
                        visible = appeared,
                        reduceMotion = reduceMotion,
                        label = "checklist$index",
                    ),
                )
            }
        }
    }
}

@Composable
private fun ChecklistRow(
    item: ReportChecklistItem,
    modifier: Modifier = Modifier,
) {
    val iconColor = when (item.state) {
        ChecklistItemState.Complete -> MMColors.Primary.copy(0.9f)
        ChecklistItemState.Incomplete -> MMColors.SemanticWarning.copy(0.85f)
        ChecklistItemState.Locked -> MMColors.TextMuted.copy(0.45f)
    }
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(vertical = 1.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(
            imageVector = when (item.state) {
                ChecklistItemState.Complete -> Icons.Filled.CheckCircle
                ChecklistItemState.Locked -> Icons.Filled.Lock
                ChecklistItemState.Incomplete -> Icons.Outlined.Circle
            },
            contentDescription = null,
            tint = iconColor,
            modifier = Modifier.size(18.dp),
        )
        Spacer(Modifier.width(10.dp))
        Column(Modifier.weight(1f)) {
            Text(
                item.title,
                color = if (item.state == ChecklistItemState.Locked) {
                    MMColors.TextMuted
                } else {
                    MMColors.TextPrimary.copy(0.92f)
                },
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
            )
            Text(item.detail, color = MMColors.TextMuted, fontSize = 12.sp)
        }
    }
}

File: movemork/Shared/MMReportReadinessChecklist.swift
//
//  MMReportReadinessChecklist.swift
//  movemork
//
//  Secondary proof checklist — lighter than the report preview hero.
//

import SwiftUI

struct MMReportReadinessChecklist: View {
    let items: [MMProofChecklistItem]
    var appeared: Bool = true

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("Proof checklist")
                .font(.system(size: 13, weight: .semibold))
                .foregroundStyle(MoveMarkTheme.Colors.textMuted)
                .padding(.leading, 2)

            VStack(spacing: 0) {
                ForEach(Array(items.enumerated()), id: \.element.id) { index, item in
                    if index > 0 {
                        Divider()
                            .background(Color.white.opacity(0.06))
                            .padding(.vertical, 6)
                    }

                    checklistRow(item)
                        .mmAppearRise(isVisible: appeared, delay: 0.04 + Double(index) * 0.03, offset: 4)
                }
            }
            .padding(.horizontal, 14)
            .padding(.vertical, 10)
            .mmProofCardSurface(.neutral, cornerRadius: 16)
        }
    }

    private func checklistRow(_ item: MMProofChecklistItem) -> some View {
        HStack(alignment: .center, spacing: 10) {
            Image(systemName: iconName(for: item.state))
                .font(.system(size: 13, weight: .semibold))
                .foregroundStyle(iconColor(for: item.state))
                .frame(width: 18, height: 18)
                .contentTransition(.symbolEffect(.replace))
                .animation(MMMotion.statusMorph, value: item.state)

            VStack(alignment: .leading, spacing: 2) {
                Text(item.title)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundStyle(
                        item.state == .locked
                            ? MoveMarkTheme.Colors.textMuted
                            : MoveMarkTheme.Colors.textPrimary.opacity(0.92)
                    )

                Text(item.detail)
                    .font(.system(size: 12, weight: .regular))
                    .foregroundStyle(MoveMarkTheme.Colors.textMuted)
                    .lineLimit(2)
            }

            Spacer(minLength: 0)
        }
        .padding(.vertical, 1)
    }

    private func iconName(for state: MMProofChecklistItem.State) -> String {
        switch state {
        case .complete: return "checkmark.circle.fill"
        case .incomplete: return "circle"
        case .locked: return "lock.fill"
        }
    }

    private func iconColor(for state: MMProofChecklistItem.State) -> Color {
        switch state {
        case .complete: return MoveMarkTheme.Colors.primary.opacity(0.9)
        case .incomplete: return MoveMarkTheme.Colors.textMuted.opacity(0.75)
        case .locked: return MoveMarkTheme.Colors.textMuted.opacity(0.45)
        }
    }
}
